### ISSUE TITLE
MISSING_CASE Quick fix Errors

### ISSUE THREAD

**Issue by MonkeyDoug** [boyland/sasylf#121]
In some instances, the quick fixes for MISSING_CASE generates other errors.

To reproduce these errors, just go to the specified line and apply the quick fix for the MISSING_CASE error. The additional errors should show up in eclipse.

bad07.slf line 271:  SASyLF Internal Error: Couldn't print (preserved [...]) for each generated case

bad09.slf line 60: parse error on the generated case

bad43.slf line 236/297: unification incomplete on the generated case

bad72.slf line 78: SASyLF Internal Error: Should not have unification error here! ...

**Comment by boyland** [comment]
Regarding bad07:  The problem is that "preserved" is a bad relation, and the print system cannot handle it.  It's not related to MISSING_CASE or the QuickFix system.

Regarding bad09: Yes, the quick fix messes up the syntax because the case analysis is all on one line.  The quick fix system assumes things are on their own line, but one can write an empty case analysis all on one line.  I knew of this limitation but felt it wasn't important to fix, but maybe I should put it higher priority?

Regarding bad43: Here the rule is written in such a way that the case for the rule will be incomplete unification.  I don't think is a problem with MISSING_CASE per se.  It's doing the best possible.  SASyLF already warned the user that the rule will lead to unification incomplete errors.

Regarding bad72: This looks like an error in SASyLF pattern matching engine. Oh.  There is another error on the same line: the context includes a variable that shouldn't be there.  So, this case is interesting, I'll investigate it more as an internal error, but it's not the "fault" of MISSING_CASE or the QuickFix system.

In summary: what I see is
* one is a bone fide error of the QuickFix system (bad09)
* one is possibly an error in the internal pattern engine, but not of the MISSING_CASE/QuickFix (bad72)
* two of them are "won't fix/invalid" because they represent cascading unfixed errors (bad07, bad43)

Do you agree?

**Comment by boyland** [comment]
Correction: There are two issues going on with bad72:
- The missing case system is outputting a spurious missing case.
- The error message of the spurious case is "internal error" when it should be redundant/unnecessary

So: bad72 is *also* relevant to MISSING_CASE (by virtue of the first problem), but on the *generation* of the quick fix, not the application.

**Comment by MonkeyDoug** [comment]
I see. So is the design of the quick fixes that even if we can introduce errors (bad07, bad43), we still apply them?

**Comment by boyland** [comment]
Sort of, yes.  The quick fix is only responsible for fixing the missing case.  For bad07: It's reasonable to assume that "proof by unproved" will not cause an error -- this is a problem with the TermPrinter, not with anything in the proof system itself, let alone case analysis.  For bad43: the problem is that no pattern matching will work since the rule is badly written. I'm OK with the quick fix generating an error -- because then at least someone can see the pattern that causes the problem.

**Comment by boyland** [comment]
I just checked in a fix of the problem that was illustrated with fixing bad72.slf.  This was a base engine problem.
